### DISCUSSION TITLE
I can't get fabric -y to work

### DISCUSSION THREAD

**Discussion by magentaFire** [danielmiessler/Fabric#1151] [Q&A]
I've been trying to `extract_wisdom` from a youtube video using `fabric -y https://www.youtube.com/watch\?v\=WxYC9-hBM_g\&t\=48s --pattern extract_wisdom` but I get a weird response:
![Captura de pantalla 2024-11-19 225223](https://github.com/user-attachments/assets/95748a0f-413d-42a1-91be-3e9b1ca35ac3)
It seems that fabric (or something else) is not correctly passing the transcript from the video correctly through the pattern; I found that running  `fabric -y https://www.youtube.com/watch\?v\=WxYC9-hBM_g\&t\=48s` without `--pattern extract_wisdom` returns the whole transcript of the video, but for some reason the patterns is not working
![Captura de pantalla 2024-11-19 225454](https://github.com/user-attachments/assets/101692c8-fcc6-41d7-b4ae-acd68a2e7cc3)

Any ideas on how to correctly configure it so I can get the  answer I'm looking for?

**Comment by mseigal** [comment]
Your script needs to be formed in parts: 1. Generate transcription | 2. Take transcription and process using Fabric AI


Note there is a pipe | in between each part so the output of part 1 becomes the input parameter for part 2. 

The correct command is as follows:

fabric -y <YouTube url> | fabric -p extract_wisdom

There may be other flags and pop parameters depending on your requirements as this will not output a file or save to the clipboard.

**Reply by magentaFire** [reply]
it didn't work for me :(
![Captura de pantalla 2024-11-22 111303](https://github.com/user-attachments/assets/d0633233-d1d9-40e1-afd1-dc57e5044777)

I'm thinking that maybe I did some mistakes during the installation so I will reinstall from scratch 
Thanks for the help @mseigal :)

**Reply by NuCl34R** [reply]
That's not it, because 'fabric -y https://www.youtube.com/watch\?v\=WxYC9-hBM_g\&t\=48s --pattern extract_wisdom' works perfectly for me
[NetworkChuck_Ai.md](https://github.com/user-attachments/files/17876683/NetworkChuck_Ai.md)

@magentaFire 
My guess is the model you are using is causing it, are you using Ollama or some local LLM? When I used Ollama at first I had incomplete outputs returned, or very poor outputs. As discussed here: [Local LLM (llama3, mixtral, gemma) short and unformatted output compared to gpt-4(o)](https://github.com/danielmiessler/fabric/discussions/504#discussioncomment-10153943) might help fixing it, it did increase the quality tremendously, but I still much prefer gpt-4o-mini.

**Reply by magentaFire** [reply]
yes! I'm using ollama with llama3, llama3.1 and phi with different results each model. I followed some sources (including the readme of this repo) to install fabric and this [video ](https://youtu.be/UbDyjIIGaxQ?si=wiPBcE8oQ5lQgdMG) explains how to run fabric with ollama.... Maybe it is not possible now with the last update but it was back then when it was the python version?
I'll take a look at the article you posted. Thanks for your response @NuCl34R  :)

**Comment by mseigal** [comment]
Just after you install you need to enter fabric -U to download latest patterns. Is your api connected correctly? What is your default model?

**Comment by mseigal** [comment]
Also, I had an issue when I first installed that my environmental variables were configured in bashrc or whatever terminal emulator shell you use such as zsh.

**Comment by magentaFire** [comment]
hey guys! thank you very much for your answers!
I finally made it work by following this article's ideas: https://medium.com/@celobusana/solving-fabric-and-local-ollama-context-issues-a-step-by-step-guide-1d67e443e27e
In the end I had to create a new model with num_ctx  20480 and set the model context length to 20480 as well in the fabric --setup 
It is VERY slow but it works now

My setup:
ryzen 9 5900HX
RTX3070 laptop (8GB VRAM)
16GB RAM

**Reply by bledburn** [reply]
I read the article, and while making a new model in Ollama, using their Modelfiles, will work... there is a much easier and more direct way to do this. In any command you craft of this nature, that involves longer YouTube video transcripts, I discovered that the transcript is sent twice in the same payload to the LLM (much as the article touches on). You can modify the context window permanently by setting the --defaultContextLength, or just for the immediate command using --modelContextLength

If you guess, and go way too far blowing open your context window, you should try dialing it back only to what you need. This won't chew up as much compute, for sure. I documented this discovery, along with a few ways to tackle this problem.

To watch discovering and immediately solving the problem, watch here: https://youtu.be/6IxC-63Owx4

To learn how to predict the tokens you are going to need, and THEN set your context window, watch here: https://youtu.be/LKFN_t94yfI
